tests/test_base_client.py

```diff
@@ -0,0 +1,137 @@
+import unittest
+import recurly
+from recurly import Resource
+from .mock_resources import MyResource, MySubResource
+from .mock_client import MockClient
+import unittest.mock as mock
+from unittest.mock import Mock, MagicMock
+
+
+def update_resource_client(success):
+    conn = MagicMock()
+    conn.request = MagicMock(return_value=None)
+    response = MagicMock()
+    if success:
+        response.status = 201
+        response.read.return_value = bytes(
+            """
+            {
+                "object": "my_resource", "prop": 123
+            }
+            """,
+            "UTF-8",
+        )
+    else:
+        response.status = 422
+        response.read.return_value = bytes(
+            """
+            {
+                "error": {
+                    "type": "validation",
+                    "message": "Year must be greater than 2000, Number is required",
+                    "params":[{"param":"year","message":"must be greater than 2000"},{"param":"number","message":"is required"}]
+                }
+            }
+            """,
+            "UTF-8",
+        )
+
+    conn.getresponse = MagicMock(return_value=response)
+
+    return mock.patch("http.client.HTTPSConnection", return_value=conn)
+
+
+def get_resource_client(success):
```

Comment:
Passing `True` returns a http connection mock that returns a 200 success.
`False` returns a `404` and error mock.